### PR TITLE
Frontend UI For matching

### LINKED ISSUE #44: F3.2.3 The service shall indicate to users how long they have been waiting for a match.

### PR DESCRIPTION
**Pull request by wr1159**
Partially resolves #44, #40, #23, #18 
| Description | SS |
|--------|--------|
| Initial View | image |
| Invalid Selection | image |
| Valid Selection | image | 
| Searching | image |

### CODE REVIEW

File: frontend/components/matching/matching-form.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Label } from "@/components/ui/label";
+
+const difficulties: string[] = ["Easy", "Medium", "Hard"];
```
Comment (jq1836) [inline frontend/components/matching/matching-form.tsx:20]: Might be best to add request to get possible values of difficulties, as well as topics. Though, for topics, we should wait till #159 is done. For the purposes of this PR, I think this is fine for now.

**Review by jq1836 — APPROVED** [review]
LGTM! Added some comments for potential issues.

File: frontend/components/matching/matching-form.tsx (hunk 1 above)
Comment (wr1159) [reply frontend/components/matching/matching-form.tsx:20]: I think better to do in another PR

File: frontend/components/matching/search-progress.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+              <Clock className="h-4 w-4" />
+              <span>
+                Wait Time: {Math.floor(waitTime / 60)}:
+                {(waitTime % 60).toString().padStart(2, "0")}
+              </span>
+            </div>
+            <span>Searching...</span>
+          </div>
+        </div>
+      </CardContent>
```
[the remark this answers is not in the document]
Comment (wr1159) [reply frontend/components/matching/search-progress.tsx:39]: Okay

File: frontend/app/app/matching/page.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import AuthPageWrapper from "@/components/auth/auth-page-wrapper";
+import FindMatch from "@/components/matching/find-match";
+import { Suspense } from "react";
+
+export default function QuestionListingPage() {
```
[the remark this answers is not in the document]
Comment (wr1159) [reply frontend/app/app/matching/page.tsx]: Updated e8cda63f3783d864b56bb5e7aa5682dff5bfc1d9

**Pull request merged**